src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Operations/OperationsCapabilityProviderTests.cs

```diff
@@ -42,28 +42,19 @@ public OperationsCapabilityProviderTests()
                 _coreFeatureOptions.Value.Returns(_coreFeatureConfiguration);
             }
 
-            [Fact]
-            public void GivenAConformanceBuilder_WhenSupportsSelectableSearchParametersIsEnabled_ThenStatusOperationIsNotAdded()
+            [Theory]
+            [InlineData(true)]
+            [InlineData(false)]
+            public void GivenAConformanceBuilder_WhenCallingOperationsCapabilityForSelectableSearchParameters_ThenStatusOperationIsAddedWhenEnabled(bool added)
             {
-                _coreFeatureConfiguration.SupportsSelectableSearchParameters = true;
-                ListedCapabilityStatement listedCapabilityStatement = new ListedCapabilityStatement();
-                OperationsCapabilityProvider provider = new OperationsCapabilityProvider(_operationsOptions, _featureOptions, _coreFeatureOptions, _urlResolver);
-                ICapabilityStatementBuilder builder = Substitute.For<ICapabilityStatementBuilder>();
-                provider.Build(builder);
+                _coreFeatureConfiguration.SupportsSelectableSearchParameters = added;
 
-                builder.Received(3).Apply(Arg.Any<Action<ListedCapabilityStatement>>());
-            }
-
-            [Fact]
-            public void GivenAConformanceBuilder_WhenSupportsSelectableSearchParametersIsDisabled_ThenStatusOperationIsNotAdded()
-            {
-                _coreFeatureConfiguration.SupportsSelectableSearchParameters = false;
-                ListedCapabilityStatement listedCapabilityStatement = new ListedCapabilityStatement();
-                OperationsCapabilityProvider provider = new OperationsCapabilityProvider(_operationsOptions, _featureOptions, _coreFeatureOptions, _urlResolver);
+                var provider = new OperationsCapabilityProvider(_operationsOptions, _featureOptions, _coreFeatureOptions, _urlResolver);
                 ICapabilityStatementBuilder builder = Substitute.For<ICapabilityStatementBuilder>();
                 provider.Build(builder);
 
-                builder.Received(2).Apply(Arg.Any<Action<ListedCapabilityStatement>>());
+                builder.Received(added ? 1 : 0)
+                    .Apply(Arg.Is<Action<ListedCapabilityStatement>>(x => x.Method.Name == nameof(OperationsCapabilityProvider.AddSelectableSearchParameterDetails)));
```

Comment:
@PTaladay - In case you're also fixing this, I think this should be more reliable for a specific operation